docs/docs/reference/dropped/class-shadowing-spec.md

```diff
@@ -0,0 +1,25 @@
+---
+layout: doc-page
+title: "Dropped: Class Shadowing - More Details"
+---
+
+Spec diff: in section [5.1.4 Overriding](https://www.scala-lang.org/files/archive/spec/2.12/05-classes-and-objects.html), add *M' must not be a class*.
+
+> Why do we want to make this change to the language?
+
+tbd
```

Comment:
@odersky Do you have any input on that? I couldn't find anything on the motivation in history...

Reply:
Shadowing is a irregular case. We don't do that for any other kind of member. In both theory and the new compiler implementation the only way to implement it would be to invent a new mangled name for each inner class, which is really clunky.